backend/src/auth/api.ts

```diff
@@ -63,17 +64,37 @@ export default fastifyPlugin(async (server, opts, next) => {
     try {
       const accessToken = spotifyApi.getAccessToken();
       const refreshToken = spotifyApi.getRefreshToken();
-
       const response = await spotifyApi.getMe();
+      const { id } = response.body;
+      let user = null;
+      await axios
+        .get(`http://localhost:6011/user/${id}`)
+        .then(function answer(res) {
+          user = res.data;
+        })
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        .catch(function failed(error) {
+          // console.log(error);
+        });
+      if (user === NOT_FOUND) {
+        await axios
+          .post('http://localhost:6011/user', response.body)
+          // eslint-disable-next-line @typescript-eslint/no-unused-vars
+          .then(function answer(res) {
+            // console.log(res);
+          })
+          // eslint-disable-next-line @typescript-eslint/no-unused-vars
+          .catch(function failed(error) {
+            // console.log(error);
+          });
+      }
       response.body.access_token = accessToken;
       response.body.refresh_token = refreshToken;
-
       reply.status(HttpStatus.OK).send(response.body);
     } catch (error) {
       request.log.error(error);
       reply.send(HttpStatus.INTERNAL_SERVER_ERROR);
     }
```

Comment:
I think there was a bit of confusion regarding async await.

Maybe @TheDadi can double check here but my suggestion would be to change this part like this:

```
    try {
      const accessToken = spotifyApi.getAccessToken();
      const refreshToken = spotifyApi.getRefreshToken();
      const spotifyResponse = await spotifyApi.getMe();

      // Check if user already exists, if not => create user
      const { id } = spotifyResponse.body;
      const userApiResponse = await axios.get(`http://localhost:6011/user/${id}`);
      if (userApiResponse.data === HttpStatus.NOT_FOUND) {
        await axios.post('http://localhost:6011/user', spotifyResponse.body);
      }

      spotifyResponse.body.access_token = accessToken;
      spotifyResponse.body.refresh_token = refreshToken;
      reply.status(HttpStatus.OK).send(spotifyResponse.body);
    } catch (error) {
      request.log.error(error);
      reply.send(HttpStatus.INTERNAL_SERVER_ERROR);
    }
```